src/signals/incident-management/containers/IncidentSplitContainer/IncidentSplitContainer.js

```diff
@@ -26,32 +27,78 @@ const getParentIncident = incident => ({
 });
 
 const IncidentSplitContainer = ({ FormComponent }) => {
-  const { data, error, get, isLoading, isSuccess, post } = useFetch();
+  const { error: errorSplit, isSuccess: isSuccessSplit, post } = useFetch();
+  const {
+    data: dataParent,
+    error: errorParent,
+    get: getParent,
+    isLoading: isLoadingParent,
+    isSuccess: isSuccessParent,
+  } = useFetch();
+  const { error: errorUpdate, patch, isSuccess: isSuccessUpdate } = useFetch();
   const { id } = useParams();
   const history = useHistory();
   const dispatch = useDispatch();
-
+  const [parentIncident, setParentIncident] = useState();
+  const [directingDepartment, setDirectingDepartment] = useState([]);
+  const departments = useSelector(makeSelectDepartments);
   const subcategories = useSelector(makeSelectSubCategories);
 
   const subcategoryOptions = useMemo(
     () => subcategories?.map(category => ({ ...category, value: category.extendedName })),
     [subcategories]
   );
 
+  const updateDepartment = useCallback(
+    name => {
+      if (!name || !departments) return;
+
+      const department = departments.list.find(d => d.code === name);
+      setDirectingDepartment(department ? [{ id: department.id }] : []);
```

Comment:
The `find` filter function could potentially return `undefined` which would result in an error on this line. Shouldn't that case be handled?

Reply:
when the `department` is not found (undefined), the `department.id` won't be evaluated but the `[]` from the `else` branch